runtime/src/iree/builtins/ukernel/arch/arm_64/CMakeLists.txt

```diff
@@ -10,7 +10,8 @@
 
 iree_add_all_subdirs()
 
-if(IREE_BUILD_COMPILER AND IREE_TARGET_BACKEND_LLVM_CPU)
+iree_compiler_targeting_iree_arch(_IREE_UKERNEL_BITCODE_BUILD_ARM_64 "arm_64")
+if(_IREE_UKERNEL_BITCODE_BUILD_ARM_64)
```

Comment:
Checking my understanding: are these "runtime/src/iree/builtins/ukernel/" files actually linked in to the _compiler_, or are they linked in to the _runtime_? If they're linked in to the runtime, we should be checking _either_ the build from source compiler settings _or_ the installed compiler settings (e.g. when cross compiling)

Reply:
The part of the build that we are looking at here, is embedding the ukernels bitcode into the compiler, not the runtime. So it is intentional that this is being skipped when `IREE_BUILD_COMPILER=OFF`. When cross-compiling to Android, this ukernel bitcode has already been built into the compiler in the host build, so it will be able to generate self-contained bytecode modules for the target.

This is a fair question though because the whole build setup is complicated around here as ukernels are potentially built into multiple things:
* Into bitcode embedded in the compiler. This is used in the llvm-cpu backend with `--iree-llvmcpu-enable-microkernels`. This is the primary delivery vehicle. This is the build code we are looking at here; it gets embedded [here](https://github.com/openxla/iree/blob/c12bf50e92d78d6203fda255f3b47cb0000728b8/runtime/src/iree/builtins/ukernel/CMakeLists.txt#L139-L157) and that is used by the compiler [here](https://sourcegraph.com/github.com/openxla/iree@c12bf50e92d78d6203fda255f3b47cb0000728b8/-/blob/compiler/src/iree/compiler/Dialect/HAL/Target/LLVMCPU/Builtins/UKernel.cpp?L20-34).
* Into stand-alone static libraries, which are themselves consumed in multiple ways:
    * They get linked into the runtime to support the VMVX backend with `--iree-vmvx-enable-microkernels`. That happens as VMVX module entry points call ukernel entry points, like [here](https://github.com/openxla/iree/blob/c12bf50e92d78d6203fda255f3b47cb0000728b8/runtime/src/iree/modules/vmvx/module.c#L594).
    * They are linked into stand-alone, [self-contained benchmarks and tests](https://github.com/openxla/iree/tree/c12bf50e92d78d6203fda255f3b47cb0000728b8/runtime/src/iree/builtins/ukernel/tools) --- so a 3rd party can work on ukernels (say, for their own hardware target) as a self-contained C project.

So that's fairly complicated, and I haven't even mentioned `experimental/cpu_ukernel' plugins as the status of that is still in flux as this is being at least partly deprecated in favor of bitcode now that we have that.